sdlib/d/gc/tcache.d

```diff
@@ -492,6 +492,25 @@ private:
 		triggerDeallocationEvent(npages * PageSize);
 	}
 
+	/**
+	 * Free directly to the arena, skip all threadCache metadata/caches.
+	 * Safe to call from a signal handler.
+	 */
+	void freeToArena(void* ptr) {
+		import d.gc.emap, d.gc.base;
+		auto emap = CachedExtentMap(&gExtentMap, &gBase);
```

Comment:
I think we could query the emap directly without creating a cache here. the cache being fresh, it's a guaranteed miss anyways, and we throw it away after, so populating it is also moot.